linkerd/telemeter/usage/src/main/scala/io/buoyant/linkerd/telemeter/UsageDataTelemeter.scala

```diff
@@ -0,0 +1,201 @@
+package io.buoyant.linkerd.telemeter
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility
+import com.fasterxml.jackson.annotation.JsonInclude.Include
+import com.fasterxml.jackson.annotation.{JsonIgnore, PropertyAccessor}
+import com.fasterxml.jackson.core.JsonFactory
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import com.fasterxml.jackson.module.scala.experimental.ScalaObjectMapper
+import com.twitter.common.metrics.Metrics
+import com.twitter.conversions.time._
+import com.twitter.finagle._
+import com.twitter.finagle.http.{Method, Request, Response}
+import com.twitter.finagle.stats.NullStatsReceiver
+import com.twitter.finagle.tracing.NullTracer
+import com.twitter.finagle.util.DefaultTimer
+import com.twitter.util._
+import io.buoyant.admin.Admin
+import io.buoyant.linkerd.Linker.{LinkerConfig, LinkerConfigStackParam}
+import io.buoyant.linkerd.protocol.HttpConfig
+import io.buoyant.linkerd.usage.{Counter, Gauge, Router, UsageMessage}
+import io.buoyant.linkerd.{Build, Linker}
+import io.buoyant.telemetry.{Telemeter, TelemeterConfig, TelemeterInitializer}
+import java.util.concurrent.atomic.AtomicBoolean
+import scala.collection.JavaConverters._
+
+private[telemeter] object UsageDataTelemeter {
+  val DefaultPeriod = 1.hour
+  val ContentType = "application/x-protobuf"
+
+  case class Client(
+    service: Service[Request, Response],
+    config: Linker.LinkerConfig,
+    orgId: Option[String]
+  ) {
+    def apply(metrics: Map[String, Number]): Future[Unit] = {
+      val msg = mkUsageMessage(config, orgId, metrics)
+
+      val req = Request(Method.Post, "/")
+      req.content = UsageMessage.codec.encodeGrpcMessage(msg)
+      req.contentType = ContentType
+      service(req).unit
+    }
+  }
+
+  def mkUsageMessage(
+    config: Linker.LinkerConfig,
+    orgId: Option[String],
+    metrics: Map[String, Number]
+  ): UsageMessage =
+    UsageMessage(
+      pid = Some(java.util.UUID.randomUUID().toString),
+      orgid = orgId,
+      linkerdversion = Some(Build.load().version),
+      containermanager = mkContainerManager,
+      osname = Some(System.getProperty("os.name")),
+      osversion = Some(System.getProperty("os.version")),
+      starttime = None, //TODO
+      routers = mkRouters(config),
+      namers = mkNamers(config),
+      counters = mkCounters(metrics),
+      gauges = mkGauges(metrics)
+    )
+
+  def mkContainerManager: Option[String] =
+    sys.env.get("MESOS_TASK_ID").map(_ => "mesos")
+
+  def mkNamers(config: Linker.LinkerConfig): Seq[String] =
+    config.namers.getOrElse(Seq()).map(_.kind)
+
+  def mkGauges(metrics: Map[String, Number]): Seq[Gauge] =
+    Seq(
+      Gauge(Some("jvm_mem"), metrics.get("jvm/mem/current/used").map(_.doubleValue())),
+      Gauge(Some("jvm/gc/msec"), metrics.get("jvm/mem/current/used").map(_.doubleValue()))
+    )
+
+  val RequestPattern = s"""^*/srv/*/requests$$""".r
```

Comment:
I think you're missing character classes before the kleenes.
`s"""^.*/srv/.*/requests$$""".r` maybe?